win32/signal.c

```diff
@@ -118,16 +119,16 @@ PHP_FUNCTION(sapi_windows_set_ctrl_handler)
 
 	if (!zend_is_callable(handler, 0, NULL)) {
 		zend_string *func_name = zend_get_callable_name(handler);
-		php_error_docref(NULL, E_WARNING, "%s is not a callable function name error", ZSTR_VAL(func_name));
+		zend_type_error("%s is not a callable function name error", ZSTR_VAL(func_name));
 		zend_string_release_ex(func_name, 0);
-		RETURN_FALSE;
+		return;
 	}
 
 	if (!SetConsoleCtrlHandler(NULL, FALSE) || !SetConsoleCtrlHandler(php_win32_signal_system_ctrl_handler, add)) {
 		zend_string *func_name = zend_get_callable_name(handler);
-		php_error_docref(NULL, E_WARNING, "Unable to attach %s as a CTRL handler", ZSTR_VAL(func_name));
+		zend_throw_error(NULL, "Unable to attach %s as a CTRL handler", ZSTR_VAL(func_name));
```

Comment:
Hmm, not sure, but I don't think this failure could be caused by a programmer error, so it's probably better to stick with warning here (the others in win32/signal.c look fine).

Reply:
OK, I'll remove this one.

Reply:
@cmb69 Could you have a short look again if your comments are really resolved? I'll merge the PR afterwards.

Reply:
Yes, LGTM now. Thanks!